hazelcast/src/main/java/com/hazelcast/client/cp/internal/datastructures/semaphore/SessionAwareSemaphoreProxy.java

```diff
@@ -147,6 +151,9 @@ public boolean tryAcquire(int permits, long timeout, TimeUnit unit) {
             } catch (WaitKeyCancelledException e) {
                 sessionManager.releaseSession(this.groupId, sessionId, permits);
                 return false;
+            } catch (Throwable t) {
+                sessionManager.releaseSession(this.groupId, sessionId, permits);
+                throw rethrow(t);
```

Comment:
We return `false` for other exceptional cases. Does a user expect an exception from this method?

Reply:
I didn't want to change the existing behavior so, I re-throw the exception here